onnxoptimizer/passes/node_hash.h

```diff
@@ -0,0 +1,178 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// ATTENTION: The code in this file is highly EXPERIMENTAL.
+// Adventurous users should note that the APIs will probably change.
+
+#pragma once
+
+#include <algorithm>
+#include <functional>
+#include <type_traits>
+#include <typeinfo>
+
+#include "onnx/onnx_pb.h"
+#include "onnxoptimizer/pass.h"
+#include "onnxoptimizer/passes/logging.h"
+#include "onnxoptimizer/passes/string_utils.h"
+
+namespace ONNX_NAMESPACE {
+namespace optimization {
+
+/// https://stackoverflow.com/questions/2590677/how-do-i-combine-hash-values-in-c0x
+inline void hash_combine(std::size_t& seed) {}
+
+template <typename T, typename... Rest>
+void hash_combine(std::size_t& seed, const T& v, Rest... rest) {
+  std::hash<T> hasher;
+  seed ^= hasher(v) + 0x9e3779b9 + (seed << 6) + (seed >> 2);
+  hash_combine(seed, rest...);
+}
+
+struct SymbolCompare {
+  bool operator()(const Symbol& lhs, const Symbol& rhs) {
+    return static_cast<uint32_t>(lhs) < static_cast<uint32_t>(rhs);
+  }
+};
+
+inline bool IsSupportedHash(const Node* n) {
```

Comment:
```suggestion
inline bool IsSupportedByCSE(const Node* n) {
```

hash 不是本质要求，参见这个 comment。
所以文件名也可以改一下，比如 cse_util.h